### PR TITLE
lib-user: Refactor MyGroups with GroupCards

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Describe your changes
- add and reorganize mock panoptes resources - users, projects, user_groups, and memberships
- refactor MyGroups with GroupCardList and GroupCards closer to [related Figma design](https://www.figma.com/file/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?node-id=192%3A1049)
- add MyGroups storybook story

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - lib-user, `yarn dev` - https://local.zooniverse.org:8080/?users=markb-panoptes/groups
  - app-root, `yarn start` - https://local.zooniverse.org:3000/users/markbouslog/groups
- Which storybook stories should be reviewed?
  - http://localhost:6008/?path=/story/components-mygroups--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-user/src/components/MyGroups/components/GroupCard/GroupCard.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  displayName = '',
+  hours = 0,
+  id = '',
+  projects = 0,
+  role = ''
+}) {
+  return (
+    <StyledListItem>
+      <StyledAnchor
+        href={`https://local.zooniverse.org:8080/?groups=${id}`}
```
Comment (mcbouslog) [inline packages/lib-user/src/components/MyGroups/components/GroupCard/GroupCard.js:54]: The `StyledAnchor` component will change per a subsequent PR that incorporates next/link and internal routing.

**Review by goplayoutside3 — COMMENTED** [review]
Looking good, but there's a few app behaviors I want to double check with you:

1. Difference between data displayed in lib-user and app-root. 

    I ran lib-user locally and viewed my own groups while signed-in to staging at https://local.zooniverse.org:8080/?users=goplayoutside3/groups. It looks like this:
    lib-user
    
    However, when I ran app-root locally (either `yarn dev` or `yarn start`) and viewed my own groups at https://local.zooniverse.org:3000/users/goplayoutside3/groups?env=staging, the same two groups are displayed, but without classifications, hours, contributors, or projects numbers:
    app-root

     It looks like the "linked" response from `/memberships` doesn't include any classifications, so something might be buggy with the request once in app-root.

2. Difference between `/memberships` data returned with admin mode on and off.

    The above described behavior applies when I used admin mode "on" in app-root, too. However, there's an additional difference where admin mode "on" returns 3 groups for my profile instead of only 2, like this:
    Screenshot 2024-03-26 at 1 08 00 PM

    Why does that happen, and is it expected? UX question - Are admin users expected to  manually refresh the app-root page when admin mode is toggled on and off?

3. Lastly, some design considerations:

    a. MyGroups component is not mobile responsive at all (yet) so just triple-checking that you plan to handle it via flexbox or otherwise.

    b. I suggest pairing `accent-1` when `neutral-1` is the text's default color. For instance: `{ light: 'neutral-1', dark: 'accent-1' }` to avoid low contrast scenarios like this:
    Screenshot 2024-03-26 at 12 57 46 PM

    c. There's a minor inconsistency with the hover effect on a GroupCard. I think it's due to padding choices. My cursor becomes a pointer on hover over the inner part of GroupCard, but not close to the edge of GroupCard. This is inconsistent with the elevation applied to the outer edge of GroupCard. See these two screenshots:
    Screenshot 2024-03-26 at 1 01 38 PM
    Screenshot 2024-03-26 at 1 01 30 PM

**Comment by mcbouslog** [comment]
1. I think there's an issue with the env query param and the stats request not reflecting the env query param. Noting for app-root, I use yarn dev with my staging user and yarn start with my production user. It should update based on the env query param, but I don't think it is. All of the displayed stats (Classifications, Hours, Contributors, and Projects) come from the ERAS request. Could you double-check yarn dev with your staging user and let me know if there are any console logs or if the ERAS requests are successful or erroring?

**Review by goplayoutside3 — APPROVED** [review]
Looks much better! I can run lib-user locally or app-root locally, see my groups, and the styling mentioned above is fixed 👍

**Pull request merged**